threeML/utils/fitted_objects/fitted_source_handler.py

```diff
@@ -143,14 +155,143 @@ def _evaluate(self):
             # scroll through the independent variables
             n_iterations = np.product(self._out_shape)
 
-            with progress_bar(n_iterations, title="Propagating errors") as p:
+            if not threeML_config['parallel']['use-parallel']:
```

Comment:
This looks overcomplicated to me.

Have you looked at the code of ParallelClient? There are functions to easily perform parallel computation by just providing the worker (they even make the progress bar for you). The code you copied from (from the contour procedures) is old, and it cannot be updated because of the inherent way a contour plot is computed.